torch/serialization.py

```diff
@@ -145,77 +144,41 @@ def persistent_id(obj):
                 warnings.warn("Couldn't retrieve source code for container of "
                               "type " + obj.__name__ + ". It won't be checked "
                               "for correctness upon loading.")
-            return (obj, source_file, source)
-        if torch.is_tensor(obj):
-            serialized_tensors[obj._cdata] = obj
-            return str(obj._cdata)
+            return ('module', obj, source_file, source)
         elif torch.is_storage(obj):
-            serialized_storages[obj._cdata] = obj
-            return str(obj._cdata)
+            storage_type = normalize_storage_type(type(obj))
+            root, offset = obj._root_storage()
+            root_key = str(root._cdata)
+            location = location_tag(obj)
+            serialized_storages[root_key] = root
+            is_view = obj._cdata != root._cdata
+
+            return ('storage', storage_type, root_key, location, root.size(), is_view, offset, obj.size())
```

Comment:
Is `root.size()` really needed in here? It's not needed if the storage is a view, and if it's not you'll save `obj.size()` anyway.